pittgoogle/alert.py

```diff
@@ -538,9 +535,17 @@ def get_key(
 
         return survey_field
 
-    def _prep_for_publish(self) -> tuple[bytes, Mapping[str, str]]:
-        """Serialize the alert dict and convert all attribute keys and values to strings."""
-        message = self.schema.serialize(self.drop_cutouts())
+    def _prep_for_publish(
+        self, serializer: Literal["json", "avro", None] = None
+    ) -> tuple[bytes, Mapping[str, str]]:
+        """Serialize the alert dict and convert all attribute keys and values to strings.
+
+        Args:
+            serializer (str or None, optional):
+                Whether to serialize the dict using Avro or JSON. If not None, this will override
+                :meth:`pittgoogle.Alert.schema.serializer` and is subject to the same requirements.
+        """
+        message = self.schema.serialize(self.drop_cutouts(), serializer=serializer)
```

Comment:
@hernandezc1 makes a good point that we will not always want to drop the cutouts. We should take this `drop_cutouts` call out of here and have the JSON serializer handle bytes values at the same time that it is handling NaNs, etc.

Reply:
I have solved this by adding a `drop_cutouts` kwarg to `Topic.publish` (bool; default False). Since bytes actually can be json serialized but require special handling, I've adding the special handling to the json serializer so that if we do want to pass bytes objects on at some point, we can. (We choose to drop the cutouts because they are large.)